sbi/neural_nets/classifier.py

```diff
@@ -0,0 +1,170 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Affero General Public License v3, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import torch
+from pyknos.nflows.nn import nets
+from torch import Tensor, nn, relu
+
+from sbi.utils.sbiutils import standardizing_net
+
+
+def build_input_layer(
+    batch_x: Tensor = None,
+    batch_y: Tensor = None,
+    z_score_x: bool = True,
+    z_score_y: bool = True,
+) -> nn.Module:
+    """Builds input layer for classifers that optionally z-scores
+
+    Args:
+        batch_x: Batch of xs, used to infer dimensionality and (optional) z-scoring
+        batch_y: Batch of ys, used to infer dimensionality and (optional) z-scoring
+        z_score_x: Whether to z-score xs passing into the network
+        z_score_y: Whether to z-score ys passing into the network
+        hidden_features: Number of hidden features
+
+    Returns:
+        Input layer that optionally z-scores
+    """
+    if z_score_x:
+        embedding_net_x = standardizing_net(batch_x)
+    else:
+        embedding_net_x = nn.Identity()
+
+    if z_score_y:
+        embedding_net_y = standardizing_net(batch_y)
+    else:
+        embedding_net_y = nn.Identity()
+
+    class StandardizeInputs(nn.Module):
+        def __init__(self, embedding_net_x, embedding_net_y):
+            super().__init__()
+            self.embedding_net_x = embedding_net_x
+            self.embedding_net_y = embedding_net_y
+
+        def forward(self, t):
+            numel = t[0].numel()
+            out = torch.cat(
+                [
+                    self.embedding_net_x(t[:, : numel // 2]),
+                    self.embedding_net_y(t[:, numel // 2 :]),
```

Comment:
This indexing confuses me. Does it assume that x and y have the same number of elements?

Reply:
Good catch, resolved